pkg/cache/cache.go

```diff
@@ -175,9 +183,8 @@ func (c *calicoCache) primeCache() error {
 		return err
 	}
 
-	for _, profile := range etcdObjList {
-		calicoKey := c.KeyFunc(profile)
-		c.Prime(calicoKey, profile)
+	for name, etcdObj := range etcdObjList {
```

Comment:
Looks like we're eliminating the need for `KeyFunc`, so I think we can just get rid of that altogether?

I think it's still valuable to have a KeyFunc defined per-resource so that the keys that are used are consistent.  Maybe that belongs in the conversion interface for each resource.

Basically there are two places where a key needs to be generated:
- The informer `AddFunc`
- The resource cache `ListFunc`

The best way to make sure they generate the same key is to share a helper function.